### PR TITLE
feat(ui): Add preferred external id keys for projects, components and releases

### PR DESCRIPTION
**Pull request by maierthomas**
- keys can be configured in the etc/sw360 properties file
- project configuration: project.externalkeys
- component configuration: component.externalkeys
- release configuration: release.externalkeys

closes eclipse/sw360#221

### CODE REVIEW

**Comment by maierthomas** [comment]
_Needs clarification: Which keys should be available for each sw360 object_
- project
- component
- release

**Comment by mcjaeger** [comment]
release:
* com.github.id
* org.maven.id
component:
* com.github.id
projects:

**Comment by maxhbr** [comment]
Maybe
- [purl](https://github.com/package-url/purl-spec)
- maven identifier
- com.gitlab.id

It might be also an Idea to move the cpeId to this list, since it is actually just an external ID.

**Comment by maierthomas** [comment]
added followings keys
- org.maven.id, com.github.id, com.gitlab.id, purl.id

removed keys for projects (datalist for project is emtpy)

File: frontend/sw360-portlet/src/main/webapp/html/utils/includes/editExternalIds.jsp [hunk 1]
```diff
@@ -36,25 +36,26 @@
...
                     var key = "", value = "";
-                }
+            }
+
             var newRowAsString =
                 '<tr id="' + rowId + '" class="bodyRow">' +
                 '<td width="46%">' +
-                '<input class="keyClass" id="<%=PortalConstants.EXTERNAL_ID_KEY%>' + rowId + '" name="<portlet:namespace/><%=PortalConstants.EXTERNAL_ID_KEY%>' + rowId + '" required="" minlength="1" class="toplabelledInput" placeholder="Enter external id key" title="Input name" value="' + key + '"/>' +
+                '<input list="externalKeyList" class="keyClass" id="<%=PortalConstants.EXTERNAL_ID_KEY%>' + rowId + '" name="<portlet:namespace/><%=PortalConstants.EXTERNAL_ID_KEY%>' + rowId + '" required="" minlength="1" class="toplabelledInput" placeholder="Enter external id key" title="external id name" value="' + key + '"/>' +
+                prepareKeyDatalist(listKeys) + // creates a datalist with preferred key names
```
[the remark this answers is not in the document]
Comment (kallesoranko) [reply frontend/sw360-portlet/src/main/webapp/html/utils/includes/editExternalIds.jsp]: ok, fine if it was intended to be there

**Comment by adrian-evo** [comment]
For projects and components, changing of external ID through moderation is not working. Proposed changes are empty.

For releases it is working.

mc

**Pull request merged**